templates/Xamarin/Features/TelemetryHockeyApp.Prism/Param_ProjectName/Services/TrackingService.cs

```diff
@@ -0,0 +1,295 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xamarin.Forms;
+
+namespace Param_ItemNamespace.Services
+{
+    public class TrackingService : ITrackingService
+    {
+        public TrackingLoggingLevel AppLoggingLevel { get; set; } = TrackingLoggingLevel.Information;
+        public TrackingLoggingLevel TrackEventLogLevel { get; set; } = TrackingLoggingLevel.Information;
+        public TrackingLoggingLevel TrackDependencyLogLevel { get; set; } = TrackingLoggingLevel.Information;
+        public TrackingLoggingLevel TrackPageViewLogLevel { get; set; } = TrackingLoggingLevel.Information;
+        public TrackingLoggingLevel TrackExceptionLogLevel { get; set; } = TrackingLoggingLevel.Error;
+        public TrackingLoggingLevel TrackMetricLogLevel { get; set; } = TrackingLoggingLevel.Warning;
+        public int AggregateMetricLoggingDelay { get; set; } = 60000;
+        protected readonly Dictionary<string, TrackingServiceMetric> TrackingServiceMetricCollection = new Dictionary<string, TrackingServiceMetric>();
+        public BackgroundWorker thread;
+        protected readonly Boolean stop = false;
+        protected readonly object _lockObject = new object();
+        protected ITrackingAPI TrackingAPI { get; private set; }
+
+        public TrackingService(ITrackingAPI trackingAPI)
+        {
+            TrackingAPI = trackingAPI;
+            thread = new BackgroundWorker();
+            thread.DoWork += async (o, e) =>
+            {
+                while (!stop)
```

Comment:
For the background thread, this is used to send metrics on an interval in case developers are tracking metrics on a high volume API – this was used for NASCAR logging GPS data every few seconds. So it runs continuously. Perhaps there should be a mechanism to disable this altogether if perf is a concern

Reply:
No, I'm not concerned with performance per se.  If the app was suspended, would this background worker still run?  I don't think it would, but just wondering.